files/scripts/update-cluster.sh

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env bash
+set -eu
+
+function patchCluster {
+  local configLocation=$1
+  local authToken=$2
+  echo "Patching cluster"
+  waitService "http://graphdb-cluster-proxy:7200/proxy/ready" "$authToken"
+  curl -o patchResponse.json -isSL -m 15 -X PATCH  --header "Authorization: Basic ${authToken}" --header 'Content-Type: application/json' --header 'Accept: application/json' -d @"$configLocation" 'http://graphdb-cluster-proxy:7200/rest/cluster/config'
+    if grep -q 'HTTP/1.1 200' "patchResponse.json"; then
+      echo "Patch successful"
+    elif grep -q 'Cluster does not exist.\|HTTP/1.1 412' "patchResponse.json" ; then
+      echo "Cluster does not exist"
+    else
+      echo "Cluster patch failed, received response:"
+      cat patchResponse.json
+      echo
+      exit 1
+    fi
+}
+
+function removeNodes {
+  local expectedNodes=$1
+  local authToken=$2
+  local namespace=$3
+  local currentNodes=$(getNodeCountInCurrentCluster "$authToken")
+  local nodes=""
+# if there is no cluster or current nodes are less or equal to expected so no need to remove more, exit
+  if [ "$currentNodes" -lt 2 ] || [ "$currentNodes" -le "$expectedNodes" ]; then
+    echo "No scaling required"
+    exit 0
+  fi
+# if there is a cluster and we wanna scale to 1 node, delete it (we would have exit on the last if in case on no cluster)
+  if [ "$expectedNodes" -lt 2 ]; then
+    echo "Deleting cluster"
+    deleteCluster "$authToken"
+    exit 0
+  fi
+
+  for ((i = expectedNodes; i < currentNodes; i++)) do
+    nodes=${nodes}\"graphdb-node-$i.graphdb-node.${namespace}.svc.cluster.local:7300\"
+    if [ $i -lt $(expr $currentNodes - 1) ]; then
+      nodes=${nodes}\,
+    fi
+  done
+  nodes=\{\"nodes\":\[${nodes}\]\}
+  waitService "http://graphdb-cluster-proxy:7200/proxy/ready" "$authToken"
+  curl -o clusterRemove.json -isSL -m 15 -X DELETE --header 'Content-Type: application/json' --header 'Accept: application/json' --header "Authorization: Basic ${authToken}" -d "${nodes}"  'http://graphdb-cluster-proxy:7200/rest/cluster/config/node'
+  if grep -q 'HTTP/1.1 200' "clusterRemove.json"; then
+    echo "Scaling down successful."
+  else
+    echo "Issue scaling down:"
+    cat clusterRemove.json
+    echo
+    exit 1
+  fi
+}
+
+function addNodes {
+  local expectedNodes=$1
+  local authToken=$2
+  local namespace=$3
+  local timeout=$4
+  local currentNodes=$(getNodeCountInCurrentCluster "$authToken")
+  local nodes=""
+# if there is no cluster or current nodes are more or equal to expected so no need to add more, exit
+  if [ "$currentNodes" -lt 2 ] || [ "$currentNodes" -ge "$expectedNodes" ]; then
+    echo "No scaling required"
```

Comment:
why not create it if scaling from 1 node to more? or the flow will go to a create cluster function in this case?

Reply:
the weight value of the create cluster job is lower so it will execute first